### ISSUE TITLE
MarginLoss produces incorrect result when both num_pos_pairs and num_neg_pairs are 0

### ISSUE THREAD

**Issue by jac99** [scikit-learn-contrib/metric-learn#287]
It may happen, that late in the training process both positive and negative distances in all triplets in some batch fall below thresholds. In such case MarginLoss produces very large loss (like 997187911680.0).  This breaks computation of batch statistics, such as a mean loss per batch, which I include in my code.

I looked into MarginLoss code and this is because when pair_count (number of active pairs) becomes zero,  total loss is equal to beta_reg_loss divided by 1e-16. Which produces a very large number. This is done by below piece of code:

```
        pair_count = self.num_pos_pairs + self.num_neg_pairs 

        return (torch.sum(pos_loss + neg_loss) + beta_reg_loss) / (pair_count + 1e-16)
```

When pair_count is zero, I think it's more logical to return loss equal to zero. Not some very large number. Maybe MarginLoss code could be amended to something like:

```
        return (torch.sum(pos_loss + neg_loss) + beta_reg_loss) / (max(pair_count,1))
```
This would prevent returning very large loss, when pair_count is zero. Or maybe more appropriate would be:
```
        if pair_count >= 1:
              return (torch.sum(pos_loss + neg_loss) + beta_reg_loss) / pair_count
       else:
              return ...grad enabled tensor set to zero...
```

**Comment by perimosocordiae** [comment]
I'm not familiar with the code you're quoting here. Perhaps you opened this issue on the wrong repo?

**Comment by jac99** [comment]
Ups, sorry, wrong window. Please remove this issue.
